Plugin.pm

```diff
@@ -282,12 +282,58 @@ sub trackInfoMenu {
 	my $extid = $track->extid;
 	$extid ||= $url if $url =~ /^tidal:/;
 
-	return _objInfoMenu($client,
-		$extid,
-		$track->artistName || $remoteMeta->{artist},
-		$track->album ? $track->albumname : $remoteMeta->{album},
-		$track->title || $remoteMeta->{title},
-	);
+	my $artist = $track->remote ? $remoteMeta->{artist} : $track->artistName;
+	my $album  = $track->remote ? $remoteMeta->{album} : ( $track->album ? $track->album->name : undef );
+	my $title  = $track->remote ? $remoteMeta->{title} : $track->title;
+
+	my $search = cstring($client, 'SEARCH');
+	my $items = [];
+	push @$items, {
+		name => "$search " . cstring($client, 'ARTISTS') . " '$artist'",
+		type => 'link',
+		url => \&search,
+		image => 'html/images/artists.png',
+		passthrough => [ {
+			type => 'artists',
+			query => $artist,
+		} ],
+	} if $artist;
+
+	push @$items, {
+		name => "$search " . cstring($client, 'ALBUMS') . " '$album'",
+		type => 'link',
+		url => \&search,
+		image => 'html/images/albums.png',
+		passthrough => [ {
+			type => 'albums',
+			query => $album,
+		} ],
+	} if $album;
+
+	push @$items, {
+		name => "$search " . cstring($client, 'SONGS') . " '$title'",
+		type => 'link',
+		url => \&search,
+		image => 'html/images/playall.png',
+		passthrough => [ {
+			type => 'tracks',
+			query => $title,
+		} ],
+	} if $title;
+
+	my $tid = $track->remote ? $remoteMeta->{id} : undef;
```

Comment:
Oh, I mentioned this part in the overall review comment: no need to search online when there's a TIDAL ID. Great start here! On a local track object this could be extracted `$track->extid` ("external ID"), which would be something like `tidal://50767187.flc`. `Plugins::TIDAL::ProtocolHandler::_getId($track->extid)` would extract the actual ID.

Reply:
That's what I tried first but `$track->extid` is empty here.  

I think the `$remoteMeta->{id}` must be limited to "tidal" tracks as this menu shows up on all tracks.  But the `id` is only useable when it comes from TIDAL.  

I just try `$track->url`.